docs/user_guide/2_software_description.md

```diff
@@ -1042,3 +1042,145 @@ Next, go to the 'Stored States' tab in 'Motion Planning'. Here you have full con
 .. image:: ../img/save_state.png
 ```
 
+## Creating a new world/scene
+
+In this section, instructions on how to create, modify and save new `.world` and `.scene` file are provided.
+
+**NOTE:** The instructions below require properly setup [sr_robot_launch](https://github.com/shadow-robot/sr_interface/tree/kinetic-devel/sr_robot_launch) package.
+
+### Running template world file
+
+In order to start creating a new world file, first you need to run a launch file with a template world file , i.e.:
+
+```sh
+roslaunch sr_world_generator create_world_template.launch
+```
+This will open Gazebo and Rviz with a robot in place:
+
+<p align="center">
+  <img src="../img/empty_world.png" width="400"/>
+</p>
+
+In most cases, when one of Shadow's robot tables is used, the above command will suffice. However, the launch file can be run with multiple arguments, i.e.:
+
+```sh
+roslaunch sr_world_generator create_world_template.launch arg_0:=value_for_arg_0 arg_1:=value_for_arg_1 ...
+```
+
+Arguments available for the launch file:
+* **start_home** - if set to `true`, robot will start in a predefined home pose. Default value: `true`
+* **scene** - if set to `true`, a scene from world file defined by the world argument will be generated. Default value: `false`
+* **initial_z** - value defining positioning of the robot base in the world frame. Default value: `0.7751`
+* **world** - path to a world file that will be generated in gazebo and rviz after running the launch file. No default value, needs to be explicitly specified if scene is set to `true`.
+
+As an example, a launch file starting with robot NOT in home position with a base at 0.5m height would be called as follows:
+
+```sh
+roslaunch sr_world_generator create_world_template.launch start_home:=false initial_z:=0.5
+```
+
+### Adding objects to the world
+
+In order to add existing object to the world, navigate to the left hand side bar in Gazebo and click **Insert** tab:
+
+<p align="center">
+  <img src="../img/insert_object.png" width="400"/>
+</p>
+
+A list of objects will appear. Please do not use other objects that ones kept in sr_description_common (second drop down on the list):
+
+<p align="center">
+  <img src="../img/object_list.png" width="400"/>
+</p>
+
+In order to add an object to the list, click on its name and move the cursor back to the scene. A shadow of the object will appear that you can move around. Single left click will put the object in a specified location.
+
+<p align="center">
+  <img src="../img/table_added.png" width="400"/>
+</p>
+
+In order to move the object around, click the following icon found at the top of the panel,
+
+<p align="center">
+  <img src="../img/move_object.png" width="400"/>
+</p>
+
+then click on the object. You can move it around now. It is usually easier to use the appearing axis frame instead of trying to drag the object itself.
```

Comment:
Maybe we can add that the pose can also be modified with numbers in the pose field

Reply:
fixed in cdb1fcf